### PR TITLE
fix: Issue #1052

### PR DESCRIPTION
**Pull request by yashvesikar**
This PR is to address #1052 where an empty array for requiredNamespaces returns all namespaces instead of none. 
This is my first PR so please let me know what is missing and what I can add to get this merged in 😄

### CODE REVIEW

**Comment by yashvesikar** [comment]
@isaachinman I don't see a good way to simplify the function signature for `serverSideTranslations` to pass the codeclimate check. Please let me know if you have any suggestions!

File: src/serverSideTranslations.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export const serverSideTranslations = async (
     initialI18nStore[lng] = {}
   })
 
-  if (namespacesRequired.length === 0) {
+  if (!namespacesRequired) {
```
Comment (isaachinman) [inline src/serverSideTranslations.ts:80]: Should probably be:

```tsx
!Array.isArray(namespacesRequired)
```

File: src/serverSideTranslations.ts [hunk 2]
```diff
@@ -86,6 +86,8 @@ export const serverSideTranslations = async (
       .map(locale => getLocaleNamespaces(path.resolve(process.cwd(), `${localePath}/${locale}`)))
 
     namespacesRequired = flatNamespaces(namespacesByLocale)
+  } else if (namespacesRequired.length === 0) {
```
Comment (isaachinman) [inline src/serverSideTranslations.ts:89]: What is the point of this condition?

File: src/serverSideTranslations.ts (hunk 1 above)
Comment (yashvesikar) [reply src/serverSideTranslations.ts]: Addressed

**Review by isaachinman — APPROVED** [review]

**Comment by isaachinman** [comment]
Thanks @yashvesikar!

**Comment by yashvesikar** [comment]
@isaachinman is there a date you plan to release 8.5.2? Just wondering when we can expect this fix in npm. Thanks!

**Comment by isaachinman** [comment]
Published now – sorry for the delay.

**Pull request merged**